src/main/java/org/broadinstitute/hellbender/tools/exome/segmentation/ClusteringGenomicHMMSegmenter.java

```diff
@@ -0,0 +1,305 @@
+package org.broadinstitute.hellbender.tools.exome.segmentation;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.apache.commons.math3.analysis.integration.SimpsonIntegrator;
+import org.apache.commons.math3.analysis.integration.UnivariateIntegrator;
+import org.apache.commons.math3.special.Gamma;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.broadinstitute.hellbender.tools.exome.ModeledSegment;
+import org.broadinstitute.hellbender.utils.*;
+import org.broadinstitute.hellbender.utils.hmm.ForwardBackwardAlgorithm;
+import org.broadinstitute.hellbender.utils.hmm.ViterbiAlgorithm;
+
+import java.util.*;
+import java.util.function.Function;
+import java.util.stream.IntStream;
+
+/**
+ * Segmentation and parameter learning corresponding to ClusteringGenomicHMM
+ *
+ * @author David Benjamin &lt;davidben@broadinstitute.org&gt;
+ */
+public abstract class ClusteringGenomicHMMSegmenter<T> {
+    protected final Logger logger = LogManager.getLogger(ClusteringGenomicHMM.class);
```

Comment:
i'm not sure if i understand the access modifiers for all of the fields here...why are some things protected and some private?